builder/amazon/ebs/step_cleanup_volumes.go

```diff
@@ -0,0 +1,117 @@
+package ebs
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/builder/amazon/common"
+	"github.com/mitchellh/packer/packer"
+)
+
+// stepCleanupVolumes cleans up any orphaned volumes that were not designated to
+// remain after termination of the instance. These volumes are typically ones
+// that are marked as "delete on terminate:false" in the source_ami of a build.
+type stepCleanupVolumes struct {
+	BlockDevices common.BlockDevices
+}
+
+func (s *stepCleanupVolumes) Run(state multistep.StateBag) multistep.StepAction {
+	// stepCleanupVolumes is for Cleanup only
+	return multistep.ActionContinue
+}
+
+func (s *stepCleanupVolumes) Cleanup(state multistep.StateBag) {
+	ec2conn := state.Get("ec2").(*ec2.EC2)
+	instanceRaw := state.Get("instance")
+	var instance *ec2.Instance
+	if instanceRaw != nil {
+		instance = instanceRaw.(*ec2.Instance)
+	}
+	ui := state.Get("ui").(packer.Ui)
+	amisRaw := state.Get("amis")
+	if amisRaw == nil {
+		ui.Say("No AMIs to cleanup")
+		return
+	}
+
+	if instance == nil {
+		ui.Say("No volumes to clean up, skipping")
+		return
+	}
+
+	ui.Say("Cleaning up any extra volumes...")
+
+	save := make(map[string]bool)
```

Comment:
Since you don't use the value, `struct{}` is actually more memory efficient. In practice here it doesn't matter but its genreally a good practice. Example:

```
save := make(map[string]struct{})
save["foo"] = struct{}{}
```